docs/doc/60-contributing/03-rfcs/20220704-generate-presigned-url.md

```diff
@@ -0,0 +1,179 @@
+
+Besides, `PRESIGN` can reduce the expanse of networking. All traffic goes through from the user side to s3 directly, with no extra cost for databend instances.
+
+# Guide-level explanation
+
+Users can generate a URL for reading:
+
+```sql
+MySQL [(none)]> PRESIGN @my_stage/books.csv
++--------+---------+---------------------------------------------------------------------------------+
+| method | headers | URL                                                                             |
++--------+---------+---------------------------------------------------------------------------------+
+| GET    | []      | https://example.s3.amazonaws.com/books.csv?X-Amz-Algorithm=AWS4-HMAC-SHA256&... |
++--------+---------+---------------------------------------------------------------------------------+
+```
+
+By default, the presigned URL will be expired in 1 hour. Users can specify the expiry time to 2 hours like the following:
+
+```sql
+MySQL [(none)]> PRESIGN @my_stage/books.csv EXPIRE=7200
++--------+---------+---------------------------------------------------------------------------------+
+| method | headers | URL                                                                             |
++--------+---------+---------------------------------------------------------------------------------+
+| GET    | []      | https://example.s3.amazonaws.com/books.csv?X-Amz-Algorithm=AWS4-HMAC-SHA256&... |
++--------+---------+---------------------------------------------------------------------------------+
+```
+
+A presigned URL is generated for the `read` action by default. Users can create a URL for `write` like the following:
+
+```sql
+MySQL [(none)]> PRESIGN @my_stage/books.csv EXPIRE=7200 ACTION=WRITE
++--------+---------+---------------------------------------------------------------------------------+
+| method | headers | URL                                                                             |
++--------+---------+---------------------------------------------------------------------------------+
+| PUT    | []      | https://example.s3.amazonaws.com/books.csv?X-Amz-Algorithm=AWS4-HMAC-SHA256&... |
++--------+---------+---------------------------------------------------------------------------------+
+```
+
+If the `headers` returned by `presign` is not empty, users should include them in the actual requests.
+
+```sql
+MySQL [(none)]> PRESIGN @my_stage/books.csv ACTION=WRITE
++--------+--------------------------+---------------------------------------------------------------------------------+
+| method | headers                  | URL                                                                             |
++--------+--------------------------+---------------------------------------------------------------------------------+
+| PUT    | [['x-amz-key', 'value']] | https://example.s3.amazonaws.com/books.csv?X-Amz-Algorithm=AWS4-HMAC-SHA256&... |
```

Comment:
We may need a header format that is easy to parse.

Reply:
Any ideas about that?

Reply:
> Is it realistic to return an object like `{'x-amz-key': 'value'}`?

SGTM